tests/test_lowlevel.py

```diff
@@ -426,3 +426,29 @@ def response_handler(sock):
     assert isinstance(excinfo.value, requests.exceptions.RequestException)
     assert isinstance(excinfo.value, JSONDecodeError)
     assert r.text not in str(excinfo.value)
+
+
+@pytest.mark.parametrize(
+    "method,include,exclude",
+    (
+        (requests.get, [], [b"Content-Length:", b"Transfer-Encoding:"]),
+        (requests.post, [b"Content-Length: 0\r\n"], [b"Transfer-Encoding:"]),
+    )
+)
+def test_empty_urlencoded_form_body(method, include, exclude):
+    """Ensure we use only the specified Host header for chunked requests."""
+    close_server = threading.Event()
+    server = Server(echo_response_handler, wait_to_close_event=close_server)
+
+    with server as (host, port):
+        url = f"http://{host}:{port}/"
+        resp = method(url, data=(("a", None,),))
+        close_server.set()  # release server block
+
+    assert not resp.content.endswith(b"\r\n0\r\n\r\n")
+
+    for header in include:
+        assert header in resp.content
+
+    for header in exclude:
+        assert header not in resp.content
```

Comment:
The test case test_empty_urlencoded_form_body ensures that only the specified Host header is used for chunked requests. It creates a server, sends a request using different HTTP methods (requests.get and requests.post), and verifies the response content for the presence or absence of specific headers. The test also checks that the response content does not end with a specific pattern